pkgs/development/python-modules/kivy/default.nix

```diff
@@ -0,0 +1,101 @@
+{ stdenv, buildPythonPackage, fetchPypi, maintainers, platforms, licenses, pkgs, self }:
+
+# From the 1.10.0 documentation:
+# "This version of Kivy requires at least Cython version 0.23, and has been
+# tested through 0.25.2. Later versions may work, but as they have not been
+# tested there is no guarantee."
+
+assert stdenv.lib.versionAtLeast self.cython.version "0.23";
+
+# Rather than depending on a specific older version of Cython, we accept the
+# off-chance of breakage:
+# assert stdenv.lib.versionOlder self.cython.version "0.25.3";
+
+buildPythonPackage rec {
+  pname = "Kivy";
+  version = "1.10.0";
+  name = "${pname}-${version}";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1394zh6kvf7k5d8vlzxcsfcailr3q59xwg9b1n7qaf25bvyq1h98";
+  };
+
+  # setup.py invokes git on build but we're fetching a tarball, so
+  # can't retrieve git version. We remove the git-invocation from setup.py
+  # This patch can presumably be removed on the next version of Kivy, because the problem
+  # was addressed in https://github.com/kivy/kivy/issues/5365
+  patches = [
+    ./setup.py.patch
+  ];
+
+  buildInputs = with self; [
+    pkgconfig
+    cython
+
+    kivygarden  # See https://github.com/kivy/kivy/issues/5367 suggestion I-2
+    requests  # See https://github.com/kivy/kivy/issues/5367 suggestion I-3
+    docutils # See https://github.com/kivy/kivy/issues/5367 question Q-5
+    pygments # See https://github.com/kivy/kivy/issues/5367 question Q-6
```

Comment:
Are these 4 only needed during build time? Or during runtime? If the latter -> `propagatedBuildInputs`

Reply:
Regarding docutils: purely build time.

Regarding the rest: `setup.py` refuses to install without these packages present, but they are fundamentally optional and (in the case of kivygarden especially) it's undesirable to pollute the runtime environment too much with their presence. If someone e.g. want the `pygments` functionality at runtime, it's better to have them install this explicitly than to "pull in the world"